koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -1029,11 +1029,23 @@ def get_source_build_info(self, build_id, build_nvr):
                 )
             raise koji.BuildError(err_msg)
 
+        build_types = koji_build['extra']['typeinfo']
+        if 'image' not in build_types:
+            types_list = sorted(build_types.keys())
+            err_msg = ('koji build {} is {} build, source container '
+                       'build needs image build'.format(koji_build['nvr'], types_list))
+            raise koji.BuildError(err_msg)
+
+        elif 'sources_for_nvr' in koji_build['extra']['image']:
+            err_msg = ('koji build {} is source container build, source container can not '
+                       'use source container build image'.format(koji_build['nvr']))
+            raise koji.BuildError(err_msg)
+
         if not build_id:
             build_id = koji_build['build_id']
         if not build_nvr:
             build_nvr = koji_build['nvr']
-        component = "%s-source" % koji_build['package_name']
```

Comment:
Unrelated change

Reply:
package_name & name have same value

but since we decided in atomic-reactor to use 'name' I wanted to make it consistent

Reply:
should be a separate commit with proper description then for better git archaeology